Rakefile

```
@@ -83,16 +83,20 @@ module Helpers
     Datadog::AppSec::WAF::VERSION::BASE_STRING
   end
 
-  def libddwaf_dir(platform:)
-    libddwaf_vendor_dir.join(libddwaf_basename(platform: platform))
-  end
-
   def libddwaf_vendor_dir
     Pathname.new('vendor/libddwaf')
   end
 
-  def libddwaf_filename(platform:)
-    "#{libddwaf_basename(platform: platform)}.tar.gz"
+  def libddwaf_tarball_filename(platform:)
+    filename = "libddwaf-#{libddwaf_version}-"
+
+    filename += if Gem::Version.new(libddwaf_version) >= Gem::Version.new('1.16.0') && platform.os == 'linux'
+                  [platform.cpu, platform.os, 'musl'].compact.join('-')
+                else
+                  [platform.os, platform.version, platform.cpu].compact.join('-')
+                end
+
+    filename + '.tar.gz'
```

I hope this would be easier to read (last statement is easier to grasp)

```suggestion
    platform += if Gem::Version.new(libddwaf_version) >= Gem::Version.new('1.16.0') && platform.os == 'linux'
                  [platform.cpu, platform.os, 'musl'].compact.join('-')
                else
                  [platform.os, platform.version, platform.cpu].compact.join('-')
                end

    "libddwaf-#{libddwaf_version}-#{platform}.tar.gz"
```